src/AzureAppConfigurationImpl.ts

```diff
@@ -196,17 +240,65 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             keyValues.push([key, value]);
         }
 
-        this.#configMap.clear(); // clear existing key-values in case of configuration setting deletion
+        this.#clearLoadedKeyValues(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.#configMap.set(k, v);
         }
     }
 
+    async #clearLoadedKeyValues() {
+        for (const key of this.#configMap.keys()) {
+            if (key !== FEATURE_MANAGEMENT_KEY_NAME) {
+                this.#configMap.delete(key);
+            }
+        }
+    }
+
+    async #loadFeatureFlags() {
+        // Temporary map to store feature flags, key is the key of the setting, value is the raw value of the setting
+        const featureFlagsMap = new Map<string, any>();
+        for (const selector of this.#featureFlagSelectors) {
+            const listOptions: ListConfigurationSettingsOptions = {
+                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
+                labelFilter: selector.labelFilter
+            };
+
+            const pageEtags: string[] = [];
+            const pageIterator = listConfigurationSettingsWithTrace(
+                this.#requestTraceOptions,
+                this.#client,
+                listOptions
+            ).byPage();
+            for await (const page of pageIterator) {
+                if (page._response.status === 200) {
+                    if (page.etag) {
+                        pageEtags.push(page.etag);
+                    }
+                }
+                for (const setting of page.items) {
+                    if (isFeatureFlag(setting)) {
+                        featureFlagsMap.set(setting.key, setting.value);
+                    }
+                }
+            }
+            selector.pageEtags = pageEtags;
+        }
+
+        // parse feature flags
+        const featureFlags = Array.from(featureFlagsMap.values()).map(rawFlag => JSON.parse(rawFlag));
+
+        // feature_management is a reserved key, and feature_flags is an array of feature flags
+        this.#configMap.set(FEATURE_MANAGEMENT_KEY_NAME, { [FEATURE_FLAGS_KEY_NAME]: featureFlags });

```

Comment:
Will this overwrite older feature flags in `configMap`?

Reply:
yes, it overwrites value with key "feature_management" in the configMap.